content/blog/2023/08/30/2023-08-30-a-glimpse-of-the-future.adoc

```diff
@@ -0,0 +1,460 @@
+As some of you may know, I've got a fondness for CPU architectures that were once seen as outlandish or still raise a few eyebrows.
+I'm referring to ARM and RISC-V, of course.
+
+ARM is making strides, finding its way into everything from laptops to servers.
+So, my personal campaign to see ARM recognized as a first-class citizen has been successful.
+
+Now, I'm turning my attention to RISC-V - my next personal endeavor and hobby.
+While it's not a fresh face in the CPU crowd, it's currently experiencing a notable uptick in popularity.
+The emergence of numerous single-board computers, workstations, and servers, reminds me of the early days of ARM, but amplified.
+I've touched on this before in one of my link:/blog/2023/03/10/miniJen-and-RISC-V/#the-risc-v-quest[blog posts].
+
+Let’s explore how JDK21, Jenkins, and RISC-V might be connected, at least in the maze of my musings.
+
+== Java Version Planning for Jenkins
+
+The ink's barely dry on the plans for which versions of the JDK the Jenkins project will support this year and the next.
+What I'm about to spill is just a snippet from a proposal whipped up by author:markewaite[Mark Waite].
+For now, it's about as definitive as a weather forecast.
+
+There'll be more detailed blog posts down the line when the final decision is made and the roadmap is as clear as a bell.
+
+=== Java 11:
+
+* Java 11 OpenJDK's public support ends in late 2024.
+* Eclipse Temurin, Red Hat, and Microsoft will support Java 11 through October 2024.
+* Oracle extends support until October 2026.
+* Amazon Corretto supports Java 11 until September 2027.
+* Proposal: End-of-life for Java 11 in Jenkins is suggested for October 31, 2024.
+* Jenkins LTS support for Java 11 will end with the August 7, 2024 release.
+* Java 21 will become the required baseline starting with the September 4, 2024 LTS release.
+* Admin monitors announcing Java 11's end of life will be enabled in Jenkins weekly releases by October 3, 2023, and in the December 13, 2023 LTS release.
+
+=== Java 17:
+
+* Java 17 adoption is increasing in Jenkins.
+The Jenkins documentation now recommends Java 17 in the installation guides and in the tutorials.
+
+image:/images/post-images/2023/08/30/2023-08-30-a-glimpse-of-the-future/image3.png[image,width=839]
+
+=== Java 21:
+
+* Java 21 releases on September 19, 2023.
+* Oracle, Eclipse Temurin, Red Hat, and others plan to support Java 21.
+* Proposal: Jenkins aims to support Java 21 in weekly releases by the end of October 2023.
+
+Are we confident Jenkins will be running on top of JDK 21 next October? +
+I guess so. Thanks to the work of Stéphane Merle, JDK21 is already available at link:https://ci.jenkins.io[https://ci.jenkins.io] to build plugins. +
+You just have to add one line to your Jenkinsfile, and your plugin will be tested on JDK21.
+
+[source,groovy]
+----
+configurations: [
+[platform: 'linux', jdk: '17'],
+[platform: 'linux', jdk: '21', jenkins: '2.414'],
```

Comment:
This was a short term technique that I was using to check that 2.414 was a good choice for the LTS baseline.  Long term, we should prefer the Jenkinsfile from the Jenkins plugin archetype, though that needs some additional time to settle for Java 21.
```suggestion
[platform: 'linux', jdk: '21'],
```